situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndServiceMapTest.java

```diff
@@ -0,0 +1,229 @@
+import io.opentelemetry.proto.trace.v1.InstrumentationLibrarySpans;
+import io.opentelemetry.proto.trace.v1.ResourceSpans;
+import io.opentelemetry.proto.trace.v1.Span;
+import org.elasticsearch.action.admin.indices.refresh.RefreshRequest;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.search.SearchHits;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.junit.Assert;
+import org.junit.Test;
+
+import static org.awaitility.Awaitility.await;
+
+public class EndToEndServiceMapTest {
+
+    private final Map<String, ResourceSpans> spanIdToRS = new HashMap<>();
+    private final List<Map<String, Object>> possibleEdges = new ArrayList<>();
+    private static final List<String> serviceNames1 = Arrays.asList(
+            "ServiceA", "ServiceA", "ServiceB", "ServiceB", "ServiceC", "ServiceC", "ServiceA", "ServiceB",
+            "ServiceB", "ServiceA", "ServiceD");
+    private static final List<String> spanIds1 = Arrays.asList("100", "200", "300", "400", "500", "600", "700", "800", "900", "1000", "1100");
+    private static final List<String> parentIds1 = Arrays.asList(
+            null, "100", "200", "300", "400", "500", "100", "700", "800", "100", "1000");
+    private static final List<String> spanNames1 = Arrays.asList(
+            "FRUITS", "CALL_SERVICE_B_APPLE", "/APPLE", "CALL_SERVICE_C_ORANGE", "/ORANGE",
+            "SOME_INTERNAL", "CALL_SERVICE_B_JACKFRUIT", "/JACKFRUIT", "SOME_INTERNAL", "CALL_SERVICE_D_PEAR", "/PEAR");
+    private static final List<Span.SpanKind> spanKinds1 = Arrays.asList(
+            Span.SpanKind.SPAN_KIND_INTERNAL, Span.SpanKind.SPAN_KIND_CLIENT, Span.SpanKind.SPAN_KIND_SERVER, Span.SpanKind.SPAN_KIND_CLIENT,
+            Span.SpanKind.SPAN_KIND_SERVER, Span.SpanKind.SPAN_KIND_INTERNAL, Span.SpanKind.SPAN_KIND_CLIENT, Span.SpanKind.SPAN_KIND_SERVER,
+            Span.SpanKind.SPAN_KIND_INTERNAL, Span.SpanKind.SPAN_KIND_CLIENT, Span.SpanKind.SPAN_KIND_SERVER);
+    private static final List<String> serviceNames2 = Arrays.asList("ServiceA", "ServiceA", "ServiceB", "ServiceA", "ServiceE");
+    private static final List<String> spanIds2 = Arrays.asList("101", "201", "301", "401", "501");
+    private static final List<String> parentIds2 = Arrays.asList(null, "101", "201", "101", "401");
+    private static final List<String> spanNames2 = Arrays.asList("VEGGIES", "CALL_SERVICE_B_ONION", "/ONION", "CALL_SERVICE_E_POTATO", "/POTATO");
+    private static final List<Span.SpanKind> spanKinds2 = Arrays.asList(
+            Span.SpanKind.SPAN_KIND_INTERNAL, Span.SpanKind.SPAN_KIND_CLIENT, Span.SpanKind.SPAN_KIND_SERVER, Span.SpanKind.SPAN_KIND_CLIENT,
+            Span.SpanKind.SPAN_KIND_SERVER);
+    private static final String SERVICE_MAP_INDEX_NAME = "otel-v1-apm-service-map";
```

Comment:
nit: You could create a class or enum and manage these, the reason is we will be adding more metadata in the next versions.

```
public enum ServiceMapData {

    DATA_100("100", "serviceA", null, "FRUITS", Span.SpanKind.SPAN_KIND_INTERNAL);


    private final String spanId;
    private final String serviceName;
    private final String parentSpanId;
    private final String name;
    private final Span.SpanKind spanKind;

    ServiceMapData(String spanId, String serviceName, String parentSpanId, String name, Span.SpanKind kind) {
        this.spanId = spanId;
        this.serviceName = serviceName;
        this.parentSpanId = parentSpanId;
        this.name = name;
        this.spanKind = kind;
    }
}
```